pageserver/src/billing_metrics.rs

```diff
@@ -0,0 +1,285 @@
+// ======================================================================
+// Billing metrics
+// ======================================================================
+
+use std::sync::Arc;
+
+use anyhow;
+use tokio::sync::Mutex;
+use tracing::*;
+use utils::id::TimelineId;
+
+use crate::task_mgr;
+use crate::tenant_mgr;
+use pageserver_api::models::TenantState;
+use utils::id::TenantId;
+
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::fmt;
+use std::str::FromStr;
+
+use chrono::{DateTime, Utc};
+
+// BillingMetric struct that defines the format for one metric entry
+// i.e.
+//
+// {
+// "metric": "s3_tenant_size_bytes",
+// "type": "absolute",
+// "tenant_id": "5d07d9ce9237c4cd845ea7918c0afa7d",
+// "timeline_id": "00000000000000000000000000000000",
+// "time": ...,
+// "value": 12345454,
+// }
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub struct BillingMetric {
+    pub metric: BillingMetricKind,
+    pub metric_type: String,
```

Comment:
```suggestion
    pub metric_type: &'static str,
```

If this is always `"absolute"`, it does not need to be an allocated string.